dump/src/reader/v4/mod.rs

```diff
@@ -147,16 +149,31 @@ pub struct V4IndexReader {
 }
 
 impl V4IndexReader {
-    pub fn new(name: String, path: &Path) -> Result<Self> {
+    pub fn new(name: String, path: &Path, tasks: BufReader<File>) -> Result<Self> {
         let meta = File::open(path.join("meta.json"))?;
         let meta: DumpMeta = serde_json::from_reader(meta)?;
 
+        let mut index_tasks: Vec<Task> = vec![];
+
+        for line in tasks.lines() {
+            let task: Task = serde_json::from_str(&line?)?;
+
+            if task.index_uid.to_string() == name {
+                index_tasks.push(task)
+            }
+        }
+
         let metadata = IndexMetadata {
             uid: name,
             primary_key: meta.primary_key,
-            // FIXME: Iterate over the whole task queue to find the creation and last update date.
-            created_at: OffsetDateTime::now_utc(),
-            updated_at: OffsetDateTime::now_utc(),
+            created_at: match index_tasks.first().unwrap().events.first() {
+                Some(TaskEvent::Created(ts)) => *ts,
+                _ => OffsetDateTime::now_utc(),
+            },
+            updated_at: match index_tasks.last().unwrap().events.last() {
+                Some(TaskEvent::Created(ts)) => *ts,
+                _ => OffsetDateTime::now_utc(),
+            },
```

Comment:
```suggestion
            created_at: created_at.unwrap_or(OffsetDateTime::now_utc()),
            updated_at:  updated_at.unwrap_or(OffsetDateTime::now_utc()),
```

Once the previous change are implemented we should be able to do this